src/geometry.jl

```diff
@@ -0,0 +1,79 @@
+
+@get_units @grid_loc @with_kw struct LandGeometry{T}
```

Comment:
And now I see I conflict with the ``RiverParameters`` you defined in the ``refactor_routing`` branch in src\groundwater\boundary_conditions.jl..... I am allowed to rename yours to something else? Or go back to Geometry?

Reply:
My suggestion is to stick to `RiverParameters` for the `groundwater` part (and not to use the term `Hyporheic`) as `River` is commonly used for the boundary condition (`River Package`) and rename your struct to `ChannelParameters`. 

The name `Channel` is related to the work on standard names (external), see also the comment on the CSDMS website:

"The word `channel` seems preferable as a generic term to words like `creek`, `stream` and `river` because it doesn't have a size connotation."

Probably best to also change this also in other parts of the code, but that can be part of another PR I think.

Reply:
But wont it get too weird? Because we have ``lateral.land.land_parameter_set`` and then ``lateral.river.river_parameter_set`` so I quite like to have the word river in there... or you envision to change ``lateral.river`` to ``lateral.channel`` too?

Reply:
Yes, my suggestion was to change this at a later stage (so not part of this PR). For now, I would then revert back to using `Geometry`, and thus keep using `river` (instead of `channel`) for now. 

It would be nice if we could use these shared parameters structs also for the routing part and `SBM` vertical concept. Some of that information is now stored as part of the `network` in the `sbm` and `sbm_gwf` model concepts.

Reply:
Clear! I used ``LandGeometry`` and ``RiverGeometry`` again but left them in the parameters.jl and reading in toml via ``river_parameter_set``. Hopefully then this will be less changes for later. Else if you want I can completely revert to using a ``geometry.jl`` file and use ``river_geometry`` in the toml.

Reply:
Yes, fine to leave these structs in parameters.jl. We can check later (not this PR) how to use these structs for the other models.